tests/TestCase/Network/Email/EmailTest.php

```diff
@@ -2592,6 +2596,74 @@ public function testZeroOnlyLinesNotBeingEmptied()
     }
 
     /**
+     * testJsonSerialize()
+     *
+     * @return void
+     */
+    public function testJsonSerialize()
+    {
+        $xmlstr = <<<XML
+<?xml version='1.0' standalone='yes'?>
+<framework>
+    <name>CakePHP</name>
+    <url>http://cakephp.org</url>
+</framework>
+XML;
+
+        $this->CakeEmail->reset()
+            ->to(['cakephp@cakephp.org' => 'CakePHP'])
+            ->from('noreply@cakephp.org')
+            ->replyTo('cakephp@cakephp.org')
+            ->cc(['mark@cakephp.org', 'juan@cakephp.org' => 'Juan Basso'])
+            ->bcc('phpnut@cakephp.org')
+            ->subject('Test Serialize')
+            ->template('default', 'test')
+            ->messageId('<uuid@server.com>')
+            ->domain('foo.bar')
+            ->viewVars([
+                'users' => TableRegistry::get('Users')->get(1, ['fields' => ['id', 'username']]),
+                'file' => fopen(TEST_APP . 'config' . DS . 'empty.ini', 'r'),
```

Comment:
Here's the diff:

```diff
diff --git a/src/Network/Email/Email.php b/src/Network/Email/Email.php
index bd1c423..8d8bdc9 100644
--- a/src/Network/Email/Email.php
+++ b/src/Network/Email/Email.php
@@ -1921,10 +1921,11 @@ class Email implements JsonSerializable
         }

         // Iterate through view vars to clean up and normalize
-        array_walk_recursive($array['_viewVars'], function (&$item, $key) {
+        array_walk_recursive($array['_viewVars'], function (&$item, $key) use (&$array) {
             // Store file contents instead of resource.
             if (is_resource($item) && get_resource_type($item) == 'stream') {
-                $item = stream_get_contents($item);
+                $item = base64_encode(stream_get_contents($item));
+                $array['_encodedViewVars'][] = $key;
             }

             if ($item instanceof Exception) {
@@ -1958,10 +1959,24 @@ class Email implements JsonSerializable
      */
     public function createFromArray($config)
     {
+        $encoded = [];
+        if (!empty($config['_encodedViewVars'])) {
+            $encoded = $config['_encodedViewVars'];
+            unset($config['_encodedViewVars']);
+        }
+
         foreach ($config as $property => $value) {
             $this->{$property} = $value;
         }

+        if (!empty($this->_viewVars) && !empty($encoded)) {
+            array_walk_recursive($this->_viewVars, function (&$item, $key) use ($encoded) {
+                if (in_array($key, $encoded)) {
+                    $item = base64_decode($item);
+                }
+            });
+        }
+
         return $this;
     }
 }
```